File: scripts/Visual/ToolTip.cs
using Godot;
using System;

namespace Peg
{
    public class ToolTip : Control
    {
        // todo this is not the best tool tip but it will do for now 
        Label tipTitle;
        Label body;
        Panel panel;
        ProgramSettings programSettings;

        public override void _Ready()
        {
            tipTitle = GetNode<Label>("Panel/title");
            body = GetNode<Label>("Panel/body");
            panel = GetNode<Panel>("Panel");
            this.programSettings = ProgramSettings.Instance();
        }
        bool nullCheck()
        {
            return tipTitle != null && body != null && panel != null&&programSettings.tooltips;
        }
        public void ShowToolTip(string title, string text)
        {
            if (nullCheck())
            {
                Vector2 pos =GetViewport().GetMousePosition();
                this.SetPosition(pos);
                body.Text = text;
                tipTitle.Text = title;
                panel.Show();
            }
        }
        public void HideToolTip()
        {
            if (nullCheck())
            {
                panel.Hide();
            }
        }

    }
}

File: src/logic/tooltip.ts

import { Subscribable } from "./subscribable";

export class ToolTip extends Subscribable {
    private static instance: ToolTip;
    x: number = 0
    y: number = 0
    visible: boolean = false
    title: string = ""
    body: string = ""
    private constructor() {
        super();


    }

    public static getInstance(): ToolTip {
        if (!ToolTip.instance) {
            ToolTip.instance = new ToolTip();
        }
        return ToolTip.instance;
    }

    Show(x: number, y: number, title: string, body: string) {
        // console.log("body", body)
        this.x = x;
        this.y = y;
        this.title = title;
        this.body = body;
        this.visible = true
        this.updateSubScribers()
    }
    Hide() {
        this.x = 0;
        this.y = 0;
        this.title = "";
        this.body = "";
        this.visible = false
        this.updateSubScribers()
    }


}